From ef22ddf278431ab39924ac468ab4b31ee6e5af95 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Tue, 7 Jun 2011 15:57:13 +0200
Subject: [PATCH] disconnect: wipe out the keeps_speed time stamp

When closing a connection, the speedchecker's timestamp is now deleted
so that it cannot accidentally be used by a fresh connection on the same
handle when examining the transfer speed.

Bug: https://bugzilla.redhat.com/679709
---
 lib/url.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/lib/url.c b/lib/url.c
index 3bc8db0..9f8c2c4 100644
--- a/lib/url.c
+++ b/lib/url.c
@@ -2671,6 +2671,7 @@ CURLcode Curl_disconnect(struct connectdata *conn, bool dead_connection)
 
   conn_free(conn);
   data->state.current_conn = NULL;
+  Curl_speedinit(data);
 
   return CURLE_OK;
 }
-- 
1.7.4.4

